pkg/prometheus/promcfg.go

```diff
@@ -3087,6 +3087,83 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 		})
 	}
 
+	// DigitalOceanSDConfig
+	if len(sc.Spec.DigitalOceanSDConfigs) > 0 {
+		configs := make([][]yaml.MapItem, len(sc.Spec.DigitalOceanSDConfigs))
+		for i, config := range sc.Spec.DigitalOceanSDConfigs {
+			assetStoreKey := fmt.Sprintf("scrapeconfig/%s/%s/digitaloceansdconfig/%d", sc.GetNamespace(), sc.GetName(), i)
+			configs[i] = cg.addSafeAuthorizationToYaml(configs[i], fmt.Sprintf("scrapeconfig/auth/%s/%s/digitaloceansdconfig/%d", sc.GetNamespace(), sc.GetName(), i), store, config.Authorization)
+			configs[i] = cg.addOAuth2ToYaml(configs[i], config.OAuth2, store.OAuth2Assets, assetStoreKey)
+
+			if config.ProxyConfig != nil {
```

Comment:
This will be addressed separately https://github.com/prometheus-operator/prometheus-operator/issues/6218

Reply:
Let's fix #6218 first to avoid more churn.

Reply:
This is fixed now